res/controllers/common-hid-packet-parser.js

```diff
@@ -1,1549 +1,2310 @@
+         * @type {number}
+         */
+        this.OutputUpdateInterval = undefined;
+
+        /**
+         * Reference to HIDModifierList object
+         *
+         * @type {HIDModifierList}
+         */
+        this.modifiers = new HIDModifierList();
+
+        /**
+         * Object of scaling function callbacks by name
+         *
+         * @type {Record.<string, scalingCallback>}
+         */
+        this.scalers = {};
+
+        /**
+         * Object of engine timer IDs, by hid-parser timer IDs
+         *
+         * @type {Record.<number, number>}
+         */
+        this.timers = {};
+
+        /**
+         * Auto repeat interval default for fields, where not specified individual (in milliseconds)
+         * - Default = 100
+         *
+         * @type {number}
+         */
+        this.auto_repeat_interval = 100;
+
+        /**
+         * Deprecated: Use postProcessDelta instead
+         * (not used in any official mapping)
+         *
+         * @type {packetCallback}
+         */
+        this.processDelta = undefined;
+
+        /**
+         * Callback that is executed after parsing incoming packet
+         * (see Traktor-Kontrol-F1-scripts.js for an example)
+         *
+         * @type {packetCallback}
+         */
+        this.postProcessDelta = undefined;
+
+    }
+    /** Function to close the controller object cleanly */
+    close() {
+        for (const name in this.timers) {
+            const timer = this.timers[name];
+            if (timer === undefined) {
+                continue;
+            }
+            engine.stopTimer(timer);
+            this.timers[name] = undefined;
+        }
     }
-    field.group = "modifiers";
-    field.name = modifier;
-    this.modifiers.set(modifier);
-};
-
-/** TODO - implement unlinking of modifiers */
-HIDController.prototype.unlinkModifier = function(_group, _name, _modifier) {
-    HIDDebug("Unlinking of modifiers not yet implemented");
-};
-
-/** Link a previously declared HID control to actual mixxx control */
-HIDController.prototype.linkControl = function(group, name, m_group, m_name, callback) {
-    var field;
-    var packet = this.getInputPacket(this.defaultPacket);
-    if (packet === undefined) {
-        HIDDebug("ERROR creating modifier: input packet " + this.defaultPacket + " not found");
-        return;
+    /**
+     * Initialize our packet data and callbacks. This does not seem to
+     * work when executed from here, but we keep a stub just in case.
+     */
+    initializePacketData() {}
+    /**
+     * Return deck number from deck name. Deck name can't be virtual deck name
+     * in this function call.
+     *
+     * @param {string} group Control group name e.g. "[Channel1]"
+     * @returns {number} Number of deck
+     */
+    resolveDeck(group) {
+        if (group === undefined) {
+            return undefined;
+        }
+        const result = group.match(/\[Channel[0-9]+\]/);
+        if (!result) {
```

Comment:
I aggree, that one single implementaion would be better, but the return values are different, in case of failure. This returns undefined, while script.deckFromGroup returns 0.